ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadWork.java

```diff
@@ -47,23 +51,32 @@
   final LineageState sessionStateLineageState;
 
   public ReplLoadWork(HiveConf hiveConf, String dumpDirectory, String dbNameToLoadIn,
-      String tableNameToLoadIn, LineageState lineageState)
-      throws IOException {
+      String tableNameToLoadIn, LineageState lineageState, boolean isIncrementalDump) throws IOException {
     this.tableNameToLoadIn = tableNameToLoadIn;
     sessionStateLineageState = lineageState;
     this.dumpDirectory = dumpDirectory;
-    this.iterator = new BootstrapEventsIterator(dumpDirectory, dbNameToLoadIn, hiveConf);
-    this.constraintsIterator = new ConstraintEventsIterator(dumpDirectory, hiveConf);
     this.dbNameToLoadIn = dbNameToLoadIn;
+    if (isIncrementalDump) {
+      incrementalIterator = new IncrementalLoadEventsIterator(dumpDirectory, hiveConf);
+      this.bootstrapIterator = null;
+      this.constraintsIterator = null;
+      incrementalLoad = new IncrementalLoadTasksBuilder(dbNameToLoadIn, tableNameToLoadIn, dumpDirectory,
+              incrementalIterator, hiveConf);
+    } else {
+      this.bootstrapIterator = new BootstrapEventsIterator(dumpDirectory, dbNameToLoadIn, hiveConf);
+      this.constraintsIterator = new ConstraintEventsIterator(dumpDirectory, hiveConf);
+      incrementalIterator = null;
```

Comment:
incrementalIterator can be removed from here and add a method in IncrementalLoadTasksBuilder which says hasNext().

Reply:
i think we should not link iterator to builder as we should be able to give different iterator to builder